renderdoc/driver/vulkan/vk_acceleration_structure.cpp

```diff
@@ -68,11 +111,115 @@ void VkAccelerationStructureInfo::Release()
+
+  return size + bufferSize;
+}
+
+void VkAccelerationStructureInfo::convertGeometryData(
+    rdcarray<VkAccelerationStructureGeometryKHR> &geometry) const
+{
+  geometry.clear();
+
+  for(const VkAccelerationStructureInfo::GeometryData &g : geometryData)
+  {
+    VkAccelerationStructureGeometryDataKHR geoUnion = {};
+    switch(g.geometryType)
+    {
+      case VK_GEOMETRY_TYPE_TRIANGLES_KHR:
+      {
+        // We'll write the offset into buffer address so when FixUpReplayBDAs is called, the real
+        // base address is just added on
+        VkDeviceOrHostAddressConstKHR vData;
+        vData.deviceAddress = g.memOffset;
+
+        VkDeviceOrHostAddressConstKHR iData;
+        iData.deviceAddress = g.memOffset;
+
+        // vkGetAccelerationStructureBuildSizesKHR just checks if the transform BDA is non-null,
+        // so fudge that here
+        VkDeviceOrHostAddressConstKHR tData;
+        tData.deviceAddress = g.buildRangeInfo.transformOffset ? g.memOffset : ~0ULL;
+
+        geoUnion.triangles = {
+            VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_GEOMETRY_TRIANGLES_DATA_KHR,
+            NULL,
+            g.tris.vertexFormat,
+            vData,
+            g.tris.vertexStride,
+            g.tris.maxVertex,
+            g.tris.indexType,
+            iData,
+            tData,
+        };
+        break;
+      }
+      case VK_GEOMETRY_TYPE_AABBS_KHR:
+      {
+        VkDeviceOrHostAddressConstKHR aData;
+        aData.deviceAddress = g.memOffset;
+
+        geoUnion.aabbs = {
+            VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_GEOMETRY_AABBS_DATA_KHR,
+            NULL,
+            aData,
+            g.aabbs.stride,
+        };
+        break;
+      }
+      case VK_GEOMETRY_TYPE_INSTANCES_KHR:
+      {
+        VkDeviceOrHostAddressConstKHR iData;
+        iData.deviceAddress = g.memOffset;
+
+        geoUnion.instances = {
+            VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_GEOMETRY_INSTANCES_DATA_KHR,
+            NULL,
+            false,
+            iData,
+        };
+        break;
+      }
+      default: RDCERR("Unhandled geometry type: %d", g.geometryType); return;
+    }
+
+    geometry.push_back({VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_GEOMETRY_KHR, NULL, g.geometryType,
+                        geoUnion, g.flags});
+  }
+}
+
+rdcarray<VkAccelerationStructureBuildRangeInfoKHR> VkAccelerationStructureInfo::getBuildRanges() const
+{
+  rdcarray<VkAccelerationStructureBuildRangeInfoKHR> result;
+  result.reserve(geometryData.size());
+
+  for(const GeometryData &geom : geometryData)
+    result.push_back(geom.buildRangeInfo);
+
+  return result;
+}
+
 VulkanAccelerationStructureManager::VulkanAccelerationStructureManager(WrappedVulkan *driver)
     : m_pDriver(driver)
 {
 }
 
+void VulkanAccelerationStructureManager::Shutdown()
+{
+  // Scratch is cleaned up automatically, but we need to reset the members for the next run
+  scratch = {};
+  scratchAddressUnion.deviceAddress = 0x0;
```

Comment:
Not that it's harmful but this comment/function doesn't match up with how it's used - this is resetting values but it only happens right as replay is shutting down so this manager is never re-used after that. Is this meant to be called somewhere else?